tests/index.ts

```diff
@@ -0,0 +1,169 @@
+import * as fs from "fs";
+import * as path from "path";
+import { SetupSettings, TestTarget, TrunkDriver } from "tests/driver";
+import { extractLandingState } from "tests/utils/landing_state";
+
+export interface TestCallbacks {
+  preCheck?: (driver: TrunkDriver) => void;
+  postCheck?: (driver: TrunkDriver) => void;
+}
+
+/**
+ * If `namedTestPrefixes` are specified, checks for their existence in `dirname`. Otherwise,
+ * automatically scan `dirname` for all available test input/output pairs.
+ * @param dirname absolute path to the test subdirectory of a linter dir.
+ * @param namedTestPrefixes optional prefixes of test input/output pairs.
+ */
+const detectTestTargets = (dirname: string, namedTestPrefixes: string[]): TestTarget[] => {
+  const parentTestDirName = path.parse(dirname).name;
+  // Sort guarantees basic.in comes before basic.out
+  const testTargets = fs
+    .readdirSync(dirname)
+    .sort()
+    .reduce((accumulator: Map<string, TestTarget>, file: string) => {
+      // Check if this is an input file. If so, set it in the accumulator.
+      const inFileRegex = /(?<prefix>.+)\.in\.(?<extension>.+)$/;
+      const foundIn = file.match(inFileRegex);
+      if (foundIn && foundIn.groups?.prefix) {
+        const prefix = foundIn.groups?.prefix;
+        if (prefix && (namedTestPrefixes.includes(prefix) || namedTestPrefixes.length === 0)) {
+          const inputPath = path.join(parentTestDirName, file);
+          accumulator.set(prefix, { prefix, inputPath, outputPath: "" });
+          return accumulator;
+        }
+      }
+
+      // Check if this is an output file and its input file is in the accumulator.
+      // If so, set it in the accumulator.
+      const outFileRegex = /(?<prefix>.+)\.out\.(?<extension>.+)$/;
+      const foundOut = file.match(outFileRegex);
+      if (foundOut) {
+        const prefix = foundOut.groups?.prefix;
+        if (prefix) {
+          const maybeTarget = accumulator.get(prefix);
+          if (maybeTarget) {
+            maybeTarget.outputPath = path.join(parentTestDirName, file);
+          }
+        }
+      }
+      return accumulator;
+    }, new Map<string, TestTarget>());
+
+  return [...testTargets.values()].filter((target) => target.outputPath.length > 0);
+};
+
+/**
+ * Setup the TrunkDriver to run tests in a `dirname`.
+ * @param dirname absolute path to the test subdirectory in a linter folder.
+ * @param setupSettings configuration for the driver's repo setup. setupGit and setupTrunk default to true.
+ * @param linterName if specified, enables this linter during setup.
+ */
+export const setupDriver = (
+  dirname: string,
+  { setupGit = true, setupTrunk = true, launchDaemon = true }: SetupSettings,
+  linterName?: string,
+  // trunk-ignore(eslint/@typescript-eslint/no-empty-function)
+  { preCheck = () => {} }: TestCallbacks = {}
```

Comment:
nit, totally optional: empty functions with comments get around this lint
```suggestion
  {
    preCheck = () => {
      // noop
    },
  }: TestCallbacks = {}
```